### PR TITLE
JNG-4690 Error massages contains info about element on entity level

### PR DESCRIPTION
**Pull request by gaborflorian**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-4690" title="JNG-4690" target="_blank">TaskJNG-4690</a>  Error massages contains information about the wrong element
  </td></table>
  <br />

### CODE REVIEW

**Review by noherczeg — APPROVED** [review]

**Review by Madmat8 — APPROVED** [review]

**Review by robertcsakany — APPROVED** [review]

File: model-test/src/test/java/hu/blackbelt/judo/meta/jsl/runtime/AnnotationTests.xtend [hunk 1]
```diff
@@ -371,7 +372,7 @@ class AnnotationTests {
 
             annotation A2(string s) on service::function;
         '''.parse => [
-            m | m.assertError(JsldslPackage::eINSTANCE.annotationArgument, JslDslValidator.TYPE_MISMATCH)
+            m | m.assertError(JsldslPackage::eINSTANCE.annotationArgument, JslDslValidator.TYPE_MISMATCH,"Non-literal at annotation argument 's'. Only literals are allowed in annotation argument.")
```
Comment (bencegelei) [inline model-test/src/test/java/hu/blackbelt/judo/meta/jsl/runtime/AnnotationTests.xtend:375]: > Non-literal at annotation argument 's'

`1` is a literal. I think the expected error message is invalid. The error message should state that the type of given argument is different then the expected, String type

File: model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend [hunk 2]
```diff
@@ -865,17 +865,17 @@ class JslDslValidator extends AbstractJslDslValidator {
...
                 JsldslPackage::eINSTANCE.enumDeclaration.name)
         }
     }
 
     @Check
-    def checkEnumOridal(EnumLiteral literal) {
+    def checkEnumOrdinal(EnumLiteral literal) {
         if (literal.value > new BigInteger("9999")) {
-            error("Enumeration ordinal is greater than the maximum allowed 9999.",
+            error("Enumeration ordinal is greater than the maximum allowed 9999 at '" + literal.name + "'.",
```
Comment (bencegelei) [inline model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend:878]: Magic number 9999. Can this be extracted as a static const or sth. like a parameter?

File: model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend [hunk 3]
```diff
@@ -1087,19 +1087,19 @@ class JslDslValidator extends AbstractJslDslValidator {
     def checkTransferField(TransferFieldDeclaration field) {
         try {
             if (field.maps !== null && !TypeInfo.getTargetType(field).isCompatible(TypeInfo.getTargetType(field.maps))) {
-                error("Type mismatch. Mapping expression value does not match field type.",
+                error("Type mismatch. Mapping expression value does not match field type at '" + field.name + "'.",
```
Comment (bencegelei) [inline model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend:1090]: Would be helpful to include mismatching types as well. (Expected: X, Got: Y)

File: model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend [hunk 4]
```diff
@@ -1087,19 +1087,19 @@ class JslDslValidator extends AbstractJslDslValidator {
...
                     JsldslPackage::eINSTANCE.transferFieldDeclaration_Maps,
                     TYPE_MISMATCH)
             }
 
             if (field.reads !== null && !TypeInfo.getTargetType(field).isCompatible(TypeInfo.getTargetType(field.reads))) {
-                error("Type mismatch. Read expression value does not match field type.",
+                error("Type mismatch. Read expression value does not match field type at '" + field.name + "'.",
```
Comment (bencegelei) [inline model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend:1096]: Would be helpful to include mismatching types as well. (Expected: X, Got: Y)

File: model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend [hunk 5]
```diff
@@ -1119,13 +1119,13 @@ class JslDslValidator extends AbstractJslDslValidator {
     def checkEntityField(EntityFieldDeclaration field) {
         try {
             if (field.defaultExpression !== null && !TypeInfo.getTargetType(field).isCompatible(TypeInfo.getTargetType(field.defaultExpression))) {
-                error("Type mismatch. Default value expression does not match field type.",
+                error("Type mismatch. Default value expression does not match field type at '" + field.name + "'.",
```
Comment (bencegelei) [inline model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend:1122]: Would be helpful to include mismatching types as well. (Expected: X, Got: Y)

File: model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend [hunk 6]
```diff
@@ -1138,7 +1138,7 @@ class JslDslValidator extends AbstractJslDslValidator {
     def checkEntityIdentifier(EntityIdentifierDeclaration field) {
         try {
             if (field.defaultExpression !== null && !TypeInfo.getTargetType(field).isCompatible(TypeInfo.getTargetType(field.defaultExpression))) {
-                error("Type mismatch. Default value expression does not match identifier field type.",
+                error("Type mismatch. Default value expression does not match identifier field type at '" + field.name + "'.",
```
Comment (bencegelei) [inline model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend:1141]: Would be helpful to include mismatching types as well. (Expected: X, Got: Y)

File: model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend [hunk 7]
```diff
@@ -1152,14 +1152,14 @@ class JslDslValidator extends AbstractJslDslValidator {
     def checkEntityDerived(EntityDerivedDeclaration derived) {
         try {
             if (derived.expression !== null && !TypeInfo.getTargetType(derived).isCompatible(TypeInfo.getTargetType(derived.expression))) {
-                error("Type mismatch. Derived value expression does not match derived field type.",
+                error("Type mismatch. Derived value expression does not match derived field type at '" + derived.name + "'.",
```
Comment (bencegelei) [inline model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend:1155]: Would be helpful to include mismatching types as well. (Expected: X, Got: Y)

File: model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend [hunk 8]
```diff
@@ -1172,14 +1172,14 @@ class JslDslValidator extends AbstractJslDslValidator {
     def checkEntityQuery(EntityQueryDeclaration query) {
         try {
             if (query.expression !== null && !TypeInfo.getTargetType(query).isCompatible(TypeInfo.getTargetType(query.expression))) {
-                error("Type mismatch. Query expression does not match query type.",
+                error("Type mismatch. Query expression does not match query type at '" + query.name + "'.",
```
Comment (bencegelei) [inline model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend:1175]: Would be helpful to include mismatching types as well. (Expected: X, Got: Y)

File: model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend [hunk 9]
```diff
@@ -1193,14 +1193,14 @@ class JslDslValidator extends AbstractJslDslValidator {
     def checkQuery(QueryDeclaration query) {
         try {
             if (query.expression !== null && !TypeInfo.getTargetType(query).isCompatible(TypeInfo.getTargetType(query.expression))) {
-                error("Type mismatch. Query expression does not match query type.",
+                error("Type mismatch. Query expression does not match query type at '" + query.name + "'.",
```
Comment (bencegelei) [inline model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend:1196]: Would be helpful to include mismatching types as well. (Expected: X, Got: Y)

File: model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend [hunk 10]
```diff
@@ -1087,19 +1087,19 @@ class JslDslValidator extends AbstractJslDslValidator {
...
             if (field.reads !== null && !TypeInfo.getTargetType(field).isCompatible(TypeInfo.getTargetType(field.reads))) {
-                error("Type mismatch. Read expression value does not match field type.",
+                error("Type mismatch. Read expression value does not match field type at '" + field.name + "'.",
                     JsldslPackage::eINSTANCE.transferFieldDeclaration_Reads,
                     TYPE_MISMATCH)
             }
 
             if (field.isIsMany && !(field.referenceType instanceof TransferDeclaration)) {
-                error("Invalid collection of primitives.",
+                error("Invalid collection of primitives at '" + field.name + "'.",
```
Comment (bencegelei) [inline model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend:1102]: If a field type is not TransferDeclaration, how is it automatically primitive? I mean, can we rephrase this error message to actually state that if its a collection its referenced type is expected to be TransferDeclaration?

File: model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend [hunk 11]
```diff
@@ -1119,13 +1119,13 @@ class JslDslValidator extends AbstractJslDslValidator {
...
             if (field.defaultExpression !== null && !TypeInfo.getTargetType(field).isCompatible(TypeInfo.getTargetType(field.defaultExpression))) {
-                error("Type mismatch. Default value expression does not match field type.",
+                error("Type mismatch. Default value expression does not match field type at '" + field.name + "'.",
                     JsldslPackage::eINSTANCE.entityFieldDeclaration_DefaultExpression,
                     TYPE_MISMATCH)
             }
 
             if (field.isIsMany && !(field.referenceType instanceof EntityDeclaration)) {
-                error("Invalid collection of primitives.",
+                error("Invalid collection of primitives at '" + field.name + "'.",
```
Comment (bencegelei) [inline model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend:1128]: Same concern as before with TransferDeclaration

File: model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend [hunk 12]
```diff
@@ -1152,14 +1152,14 @@ class JslDslValidator extends AbstractJslDslValidator {
...
-                error("Type mismatch. Derived value expression does not match derived field type.",
+                error("Type mismatch. Derived value expression does not match derived field type at '" + derived.name + "'.",
                     JsldslPackage::eINSTANCE.entityDerivedDeclaration_Expression,
                     TYPE_MISMATCH,
                     JsldslPackage::eINSTANCE.dataTypeDeclaration.name)
             }
 
             if (derived.isIsMany && !(derived.referenceType instanceof EntityDeclaration)) {
-                error("Invalid collection of primitives.",
+                error("Invalid collection of primitives at '" + derived.name + "'.",
```
Comment (bencegelei) [inline model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend:1162]: Same concern as before with TransferDeclaration

File: model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend [hunk 13]
```diff
@@ -1172,14 +1172,14 @@ class JslDslValidator extends AbstractJslDslValidator {
...
-                error("Type mismatch. Query expression does not match query type.",
+                error("Type mismatch. Query expression does not match query type at '" + query.name + "'.",
                     JsldslPackage::eINSTANCE.entityQueryDeclaration_Expression,
                     TYPE_MISMATCH,
                     JsldslPackage::eINSTANCE.dataTypeDeclaration.name)
             }
 
             if (query.isIsMany && !(query.referenceType instanceof EntityDeclaration)) {
-                error("Invalid collection of primitives.",
+                error("Invalid collection of primitives at '" + query.name + "'.",
```
Comment (bencegelei) [inline model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend:1182]: Same concern as before with TransferDeclaration

File: model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend [hunk 14]
```diff
@@ -1193,14 +1193,14 @@ class JslDslValidator extends AbstractJslDslValidator {
...
-                error("Type mismatch. Query expression does not match query type.",
+                error("Type mismatch. Query expression does not match query type at '" + query.name + "'.",
                     JsldslPackage::eINSTANCE.queryDeclaration_Expression,
                     TYPE_MISMATCH,
                     JsldslPackage::eINSTANCE.queryDeclaration.name)
             }
 
             if (query.isIsMany && !(query.referenceType instanceof EntityDeclaration)) {
-                error("Invalid collection of primitives.",
+                error("Invalid collection of primitives at '" + query.name + "'.",
```
Comment (bencegelei) [inline model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend:1203]: Same concern as before with TransferDeclaration

File: model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend [hunk 15]
```diff
@@ -1226,7 +1226,7 @@ class JslDslValidator extends AbstractJslDslValidator {
     def checkEntityOperationParameter(EntityOperationParameterDeclaration parameter) {
         try {
             if (parameter.isIsMany && !(parameter.referenceType instanceof EntityDeclaration)) {
-                error("Invalid collection of primitives.",
+                error("Invalid collection of primitives at '" + parameter.name + "''.",
```
Comment (bencegelei) [inline model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend:1229]: Same concern as before with TransferDeclaration

**Pull request merged**